flink-table/flink-table-planner/src/test/java/org/apache/flink/table/runtime/stream/sql/FunctionITCase.java

```diff
@@ -444,6 +450,31 @@ private void testUserDefinedCatalogFunction(TableEnvironment tableEnv, String cr
 		tableEnv.sqlUpdate("drop table t2");
 	}
 
+	@Test
+	public void testDataTypeBasedTypeInferenceNotSupported() throws Exception {
+		thrown.expect(ValidationException.class);
+		thrown.expectMessage("The new type inference for functions is only supported in the Blink planner.");
+
+		StreamExecutionEnvironment streamExecEnvironment = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tableEnvironment = StreamTableEnvironment.create(streamExecEnvironment);
+
+		tableEnvironment.createTemporarySystemFunction("func", SimpleScalarFunction.class);
+		Table table = tableEnvironment
+			.sqlQuery("SELECT func(1)");
+		tableEnvironment.toAppendStream(table, Row.class).print();
+
+		streamExecEnvironment.execute();
+	}
+
+	/**
```

Comment:
remove comment

Reply:
We must have a comment to satisfy the checkstyle. We have the same in the Blink's version of this test.

Reply:
True, but there we really tested different kind of functions. The comment has nothing to do with the new type inference. But I don't have a string opinion there.

Reply:
Will simplify it.